core/src/main/java/com/valtech/aem/saas/core/http/response/SuggestionDataExtractionStrategy.java

```diff
@@ -0,0 +1,66 @@
+package com.valtech.aem.saas.core.http.response;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.valtech.aem.saas.api.fulltextsearch.Suggestion;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A strategy for extracting highlighting data.
+ */
+@Slf4j
+public final class SuggestionDataExtractionStrategy implements SearchResponseDataExtractionStrategy<Suggestion> {
+
+  public static final String PN_SPELLCHECK = "spellcheck";
```

Comment:
- [ ] what does PN stand for? Property Name? can you replace without using abbrevations?